### PR TITLE
Fixed #35294 -- Fixed TEXT format of QuerySet.explain() for long plans.

### PR DESCRIPTION
**Pull request by adamchainz**
# Trac ticket number

ticket-35294

# Branch description

Fixes explain bug per ticket.

# Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" **ticket flag** in the Trac system.
- [x] I have added or updated relevant **tests**.
- [x] I have added or updated relevant **docs**, including release notes if applicable.
- [x] For UI changes, I have attached **screenshots** in both light and dark modes.

### CODE REVIEW

**Comment by adamchainz** [comment]
@tolomea

File: tests/queries/test_explain.py [hunk 1]
```diff
@@ -96,6 +96,25 @@ def test_postgres_options(self):
...
+            def endswith(self, value):
+                if value == '"':
+                    return True
+                return super().endswith(value)
+
+        qs = Tag.objects.extra(
+            tables=[PretendQuoted(f"generate_series(1,1) GS{n}") for n in range(100)]
+        )
+        result = qs.explain(format="text")
+        assert result.count("\n") >= 200
```
Comment (felixxm) [inline tests/queries/test_explain.py:116]: Use unittest assertions.

**Review by felixxm — COMMENTED** [review]
@adamchainz Thanks :+1:

File: tests/queries/test_explain.py [hunk 2]
```diff
@@ -96,6 +96,25 @@ def test_postgres_options(self):
...
+
+            def endswith(self, value):
+                if value == '"':
+                    return True
+                return super().endswith(value)
+
+        qs = Tag.objects.extra(
+            tables=[PretendQuoted(f"generate_series(1,1) GS{n}") for n in range(100)]
+        )
+        result = qs.explain(format="text")
```
[the remark this answers is not in the document]
Comment (charettes) [reply tests/queries/test_explain.py:115]: If we go with this solution I'd suggest creating the union against queryset outside of the comprehension to make the test faster

```suggestion
        other = Tag.objects.order_by()
        qs = Tag.objects.filter(name="test").union(*[other for _ in range(100)])
        result = qs.explain(format="text")
        self.assertEqual(result.count(f"Seq Scan on {Tag._meta.db_table} {Tag._meta.db_table}"), 100)
```

[the remark this answers is not in the document]
Comment (adamchainz) [reply tests/queries/test_explain.py]: Thanks, that’s perfect.

File: tests/queries/test_explain.py [hunk 3]
```diff
@@ -96,6 +96,15 @@ def test_postgres_options(self):
                     option = "{} {}".format(name.upper(), "true" if value else "false")
                     self.assertIn(option, captured_queries[0]["sql"])
 
+    @unittest.skipUnless(connection.vendor == "postgresql", "PostgreSQL specific")
```
Comment (felixxm) [inline tests/queries/test_explain.py]: Other database are also affected so I made it database independent.

**Review by felixxm — COMMENTED** [review]
@adamchainz Thanks :+1: Other databases are also affected so I made the test database independent.

**Comment by adamchainz** [comment]
Fair enough on the expansion. I started with only PostgreSQL to test a specific platform, I agree the new format should work with all text-format-supporting databases.

Thanks all for the input and merge!

**Pull request merged**